pkg/apps/controller/util/util.go

```diff
@@ -0,0 +1,901 @@
+// DeploymentConfig. The controller replica count will be zero.
+// DEPRECATED: Will be replaced with external version eventually.
+func MakeTestOnlyInternalDeployment(config *appsapi.DeploymentConfig) (*api.ReplicationController, error) {
+	obj, err := MakeDeploymentV1(config)
+	if err != nil {
+		return nil, err
+	}
+	kapiv1.SetObjectDefaults_ReplicationController(obj)
+	converted, err := legacyscheme.Scheme.ConvertToVersion(obj, api.SchemeGroupVersion)
+	if err != nil {
+		return nil, err
+	}
+	deployment := converted.(*api.ReplicationController)
+	return deployment, nil
+}
+
+// LatestDeploymentNameForConfig returns a stable identifier for config based on its version.
+func LatestDeploymentNameForConfig(config *appsapi.DeploymentConfig) string {
+	return fmt.Sprintf("%s-%d", config.Name, config.Status.LatestVersion)
+}
+
+// LatestDeploymentInfo returns info about the latest deployment for a config,
+// or nil if there is no latest deployment. The latest deployment is not
+// always the same as the active deployment.
+func LatestDeploymentInfo(config *appsapi.DeploymentConfig, deployments []*v1.ReplicationController) (bool, *v1.ReplicationController) {
+	if config.Status.LatestVersion == 0 || len(deployments) == 0 {
+		return false, nil
+	}
+	sort.Sort(ByLatestVersionDescV1(deployments))
+	candidate := deployments[0]
+	return DeploymentVersionFor(candidate) == config.Status.LatestVersion, candidate
+}
+
+// ActiveDeployment returns the latest complete deployment, or nil if there is
+// no such deployment. The active deployment is not always the same as the
+// latest deployment.
+func ActiveDeployment(input []*api.ReplicationController) *api.ReplicationController {
+	var activeDeployment *api.ReplicationController
+	var lastCompleteDeploymentVersion int64 = 0
+	for i := range input {
+		deployment := input[i]
+		deploymentVersion := DeploymentVersionFor(deployment)
+		if IsCompleteDeployment(deployment) && deploymentVersion > lastCompleteDeploymentVersion {
+			activeDeployment = deployment
+			lastCompleteDeploymentVersion = deploymentVersion
+		}
+	}
+	return activeDeployment
+}
+
+// ActiveDeployment returns the latest complete deployment, or nil if there is
+// no such deployment. The active deployment is not always the same as the
+// latest deployment.
+func ActiveDeploymentV1(input []*v1.ReplicationController) *v1.ReplicationController {
+	var activeDeployment *v1.ReplicationController
+	var lastCompleteDeploymentVersion int64 = 0
+	for i := range input {
+		deployment := input[i]
+		deploymentVersion := DeploymentVersionFor(deployment)
+		if IsCompleteDeployment(deployment) && deploymentVersion > lastCompleteDeploymentVersion {
+			activeDeployment = deployment
+			lastCompleteDeploymentVersion = deploymentVersion
+		}
+	}
+	return activeDeployment
+}
+
+// DeployerPodSuffix is the suffix added to pods created from a deployment
+const DeployerPodSuffix = "deploy"
```

Comment:
that const seem to be hidden here

Reply:
one usage in the helper function below... will remove this const and just use the string.

Reply:
I like the constant - it's just that we usually declare those at the top of the file, not line 314 but maybe that was preexisting